app/src/main/java/com/github/libretube/repo/LocalFeedRepository.kt

```diff
@@ -59,8 +61,15 @@ class LocalFeedRepository : FeedRepository {
     }
 
     private suspend fun refreshFeed(channelIds: List<String>, minimumDateMillis: Long) {
+        val extractionCount = AtomicInteger()
         for (channelIdChunk in channelIds.chunked(CHUNK_SIZE)) {
             val collectedFeedItems = channelIdChunk.parallelMap { channelId ->
+                // throttle feed extraction to avoid rate limiting
+                val count = extractionCount.getAndIncrement()
+                if (count != 0 && count % BATCH_SIZE == 0) {
```

Comment:
This will likely only let one thread sleep while the other parallel ones are still working as usual. It might make sense to move this if condition to the end of the for loop, and instead of checking if `count % BATCH_SIZE == 0` we check if `count >= BATCH_SIZE` and reset it to zero when it's the case? :)